cli/src/cpp/utils/external_locators_parser.cpp

```diff
@@ -38,34 +39,37 @@ void external_locators_parser(
     bool print_usage = false;
     std::string subelement;
     std::string key;
-    // External locator list must be keyed
-    if (!extract_element_subelement_key(element, subelement, key))
-    {
-        std::cout << "ERROR: external locator <" << element << "> list must be keyed []" << std::endl;
-        print_usage = true;
-    }
 
-    // HELP command has been handled in the upper level so usage should be printed in case of wrong arguments.
+    bool keyed = extract_element_subelement_key(element, subelement, key);
     bool set_every_element = false;
-    // Not SET commands require no values at all.
-    if (values.size() != 0 && command != CommonCommands::SET && !print_usage)
+
+    // Initialize message in case of error
+    std::ostringstream message;
+    message << "external locator <" << element << "> list";
+
+    // HELP command has been handled in the upper level so usage should be printed in case of wrong arguments.
+    // Validity checks
+    // Check that the element is the expected one
+    if (((list == ExternalLocatorsList::DATAREADER_UNICAST || list == ExternalLocatorsList::DATAWRITER_UNICAST) &&
+            element != UNICAST_ELEMENT) ||
+            (list == ExternalLocatorsList::PARTICIPANT_DEFAULT_UNICAST && element != DEFAULT_UNICAST_ELEMENT) ||
+            (list == ExternalLocatorsList::PARTICIPANT_METATRAFFIC_UNICAST && element != METATRAFFIC_UNICAST_ELEMENT))
     {
-        std::cout << "ERROR: the given command does not require any value" << std::endl;
+        std::cout << "ERROR: " << element << " subelement not recognized" << std::endl;
         print_usage = true;
     }
+    print_usage = print_usage || !check_keyed(true, keyed, message.str());
+    print_usage = print_usage ||
+            (command != CommonCommands::SET && !check_arguments(0, values.size(), "the given command", true));
```

Comment:
Just an idea here. It could be nice doing something this:

Define in `utils.hpp` 
```
static const char *common_command_to_str[] =
{ "CLEAR", "PRINT", "QUERY", "SET" };
```
Define an overload (`utils.cpp`)
```
check_command_arguments(
        const CommonCommands &command,
        uint32_t expected_arguments,
        uint32_t actual_argument_number,
        std::string& element,
        bool exact)
{
    std::ostringstream cmd_element = common_command_to_str[command] << " command for " << element;
    check_arguments(expected_arguments, actual_argument_number, cmd_element.str(), exact);
}
```
```suggestion
            (command != CommonCommands::SET && !check_command_arguments(command, 0, values.size(), message.str(), true));
```
This could be applied in all the command checks.